include/devicetree/clocks.h

```diff
@@ -7,6 +7,7 @@
  * Copyright (c) 2020, Linaro Ltd.
  *
  * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileComment: IEC-61508-SIL3
```

Comment:
Provided that you can document the impact of this comment on a macro-only API like devicetree.h in a way that I can understand and enforce, I have no fundamental objections to this. I will need training, though.

However, have you considered the impact on the generated devicetree_unfixed.h header itself?

Reply:
The question here is if these kind of SPDX tags remain on the LTS vX.Y-auditable and we repeat that for each of the v(X+n).Z-auditable's from auditable to auditable, or if such a mechanism should be merged to main.

The idea of bringing that into main and into the source files are:
- maintaining such information in a "database" separate from the source files will out-date faster than when in the files
- It also indicates in which code parts special care has to be taken, not to implement features and mechanisms which are hard to impossible to certify for safety. -> trigger the safety architect

Freedom to implement anything anyhow and safety are, to my experience, somewhat contradicting.

In the end there might be a discussion required how far the community is willing to support safety on main or if safety always is a "auditable" thing only, the safety guys having to cope with what they get from each LTS.

Reply:
> The question here is if these kind of SPDX tags remain on the LTS vX.Y-auditable and we repeat that for each of the v(X+n).Z-auditable's from auditable to auditable, or if such a mechanism should be merged to main.

I can see how this is the important question for you. For me as a DT maintainer, the important question is if you can specify:

>  the impact of this comment on a macro-only API like devicetree.h in a way that I can understand and enforce

Does that make sense?

Reply:
As I wrote in the description intro (only lately), I did the marking very roughly on suspicion of the block diagram (link in description). Whether or not exactly this file should get a tag or not I do not have sufficient knowledge in the different area to judge in detail. I depend on the people that maintain those files to finally understand what the purpose of the files are and if they should get a tag or not. I will happily remove any file not required - the less the better - but we will still need to get all required.

The impact for such marked files will be:
- for Zephyr OS source code is change
  - the requirements specifications will have to be updated
  - the test specification will have to be updated
  - the module / unit test will have to be update
- for Zephyr OS tools such as dt generator, ztest, etc.
  - tool qualification needs to be repeated in next auditable
    - updating all paper work which is similar to the Zephyr OS source code but very likely not in that extend

If the said above frightens the developers to change anything, there will be no further progress in what is considered the safety scope.
If we want to do repeated pre-certified auditable tags from LTS-es we will have to accept that we need do that work.
To me the point is more, if we are able to make the changes to the safety scope a controlled process or will be surprised with what has changed when the next auditable is branched.

I guess the main impact will be that the safety architect gets an early chance to get involved and have an eye on it and once we have requirements and (safety-)tests in place that all that gets updated and does not out-date.